### PR TITLE
Added http methods to Insecure HTTP-Method Rule

### PR DESCRIPTION
**Pull request by rain0r**
Added methods PUT, DELETE, TRACE

### CODE REVIEW

File: src/org/zaproxy/zap/extension/ascanrulesBeta/InsecureHTTPMethod.java [hunk 1]
```diff
@@ -62,7 +62,10 @@
 			{
 			"TRACE",
 			"TRACK",
-			"CONNECT"
+			"CONNECT",
+			"PUT",
+			"DELETE",
```
Comment (thc202) [inline src/org/zaproxy/zap/extension/ascanrulesBeta/InsecureHTTPMethod.java]: The description of the the alert (and "extra info") should be updated to explain why these are insecure.

File: src/org/zaproxy/zap/extension/ascanrulesBeta/InsecureHTTPMethod.java [hunk 2]
```diff
@@ -62,7 +62,10 @@
...
+			"TRACE"
```
Comment (thc202) [inline src/org/zaproxy/zap/extension/ascanrulesBeta/InsecureHTTPMethod.java]: This one is already included.

File: src/org/zaproxy/zap/extension/ascanrulesBeta/InsecureHTTPMethod.java (hunk 1 above)
Comment (kingthorin) [inline src/org/zaproxy/zap/extension/ascanrulesBeta/InsecureHTTPMethod.java]: There should probably also be additional checks around PUT and DELETE. For example: If DELETE is possible put results in a 400 or 500 condition then is it actually a vulnerability? Or if it's available but only when authenticated?

File: src/org/zaproxy/zap/extension/ascanrulesBeta/InsecureHTTPMethod.java (hunk 2 above)
Comment (rain0r) [inline src/org/zaproxy/zap/extension/ascanrulesBeta/InsecureHTTPMethod.java]: I'm sorry. Fixed that.

**Comment by kingthorin** [comment]
Please refer to: https://github.com/zaproxy/zaproxy/wiki/DevGuidelines#formatting

Formatting changes should be submitted separate from code changes.

**Comment by rain0r** [comment]
Yep I read that. Was unaware that my commit was going to this PR. 
However, there will be a little bit more to do than what I initially thought, since the added methods are not supported and will throw an Exception.

**Comment by thc202** [comment]
Which exception are you seeing? It should be possible to send PUT/DELETE requests without changes.

**Comment by rain0r** [comment]
I got this one: https://github.com/zaproxy/zap-extensions/blob/beta/src/org/zaproxy/zap/extension/ascanrulesBeta/InsecureHTTPMethod.java#L287

**Comment by thc202** [comment]
I see, but the change would be just an additional check (per @kingthorin's https://github.com/zaproxy/zap-extensions/pull/758#discussion_r100086190).

**Pull request closed without merging**